message_sender/tasks.py

```diff
@@ -77,6 +79,44 @@ def run(self, metric_name, metric_value, session=None, **kwargs):
 fire_metric = FireMetric()
 
 
+class Concurrency_Limiter(object):
+    def __init__(self):
+        self.redis_server = redis.StrictRedis(host=settings.REDIS_HOST,
+                                              port=settings.REDIS_PORT,
+                                              db=settings.REDIS_DB)
+
+    def get_current_message_count(self, msg_type):
+        keys = self.redis_server.keys(pattern=msg_type + "_messages_at_*")
```

Comment:
`keys` is `O(N)`, where `N` is the total number of keys in the database, so this could be quite slow if there are a lot of things in redis.

We know what keys we're expecting (a key for the current minute, plus a key for the `N = delay // 60` previous minutes), so we could just search through those keys, and handle the case where a key doesn't exist.